### PR TITLE
Review find group

### PR DESCRIPTION
**Pull request by benjaminbollen**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/529)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @inetic

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by benjaminbollen** [comment]
the ConnectRequests are still not compatible

File: src/routing_node.rs [hunk 1]
```diff
@@ -215,11 +215,38 @@ impl<F, G> RoutingNode<F, G> where F : Interface + 'static,
...
+                // send FindGroup request before moving to Membrane
+                let find_group_msg =
+                    try!(self.construct_find_group_msg_as_client(&our_bootstrap.1));
+                ignore(cm.send(our_bootstrap.0.clone(), try!(encode(&find_group_msg))));
+
+                Some(our_bootstrap)
+            },
+            // someone tried to bootstrap to us
+            true => None
+        };
```
Comment (inetic) [inline src/routing_node.rs]: `if {} else {}` would do as well here.

File: src/routing_node.rs [hunk 2]
```diff
@@ -215,11 +215,38 @@ impl<F, G> RoutingNode<F, G> where F : Interface + 'static,
             }
         }
 
+        let our_bootstrap = match possible_first {
+            // we bootstrapped to a node
+            false => {
+                // verify bootstrap connection
+                let our_bootstrap = match self.bootstrap {
+                    Some((ref endpoint, ref opt_name)) => {
```
Comment (inetic) [inline src/routing_node.rs]: Pattern matching can go several layers deep:
```
match self.bootstrap {
    Some((ref endpoint, Some(ref name)) => {
    ....
    }
    _ => return Err(RoutingError::FailedToBootstrap)
}
```

**Pull request merged**